src/App.tsx

```diff
@@ -37,7 +40,11 @@ const Sources = lazy(() => import('@pages').then(module => ({default: module.Sou
 const Triggers = lazy(() => import('@pages').then(module => ({default: module.Triggers})));
 const GlobalSettings = lazy(() => import('@pages').then(module => ({default: module.GlobalSettings})));
 
-const App: React.FC = () => {
+export interface AppProps {
+  pluginManager: PluginManager;
```

Comment:
Actually, I'm thinking about the community plugins and I've got a struggle. There are two options we could approach it:

* Plugins may be injected only for Dashboard context
* Plugins may be injected for both Dashboard and Cloud context

if we want it only for a dashboard, we could pass `plugins: Plugin[]` here instead of `pluginManager`. This would make that implementation internal, simplifying the dependencies. `App` should then instantiate and setup `PluginManager`.

If we want to allow injecting also in the Cloud context in the future, we should pass `plugins: PluginScope` (btw, I'm not sure if this name is appropriate for that, but it's fine). Otherwise, we would call `manager.setup()` twice - in Cloud context (before Dashboard is loaded) and in Dashboard context, while it should be setup only once.

The first solution is simpler, but the second one not much harder, yet may be helpful in the future. What are your thoughts on that?

Reply:
The second solution means that we may have both plugins for Dashboard (i.e. Test Execution) and Cloud (i.e. Environment, or something that just works all the time).

Because of that, you would need to call `manager.setup()` in the Cloud's `App` too - otherwise, it would be only instantiated in Dashboard.

Reply:
now I get what you're saying. Yes, that's true. If we also aim to have cloud plugins, we would have to call `manager.setup()` twice. It's an interesting use case to have also cloud plugins, but personally I think it's way down the road. Also, it adds another layer of complexity, since these users don't have access to the cloud repo and most likely also the plugins shouldn't be publicly available. We would need to think of an import mechanism.  Only downside I see if we decide not to do it now and only much later, is that we would need to refactor all the other places where plugins are being used. **I would still stick to OSS plugins for the time being.**

Reply:
> Also, it adds another layer of complexity, since these users don't have access to the cloud repo and most likely also the plugins shouldn't be publicly available.

The Users won't need access to the Cloud repo, they only would need to know the interface.

> We would need to think of an import mechanism.

For community plugins, even for OSS Dashboard, we still need to think about the import mechanism.

> Only downside I see if we decide not to do it now and only much later, is that we would need to refactor all the other places where plugins are being used. I would still stick to OSS plugins for the time being.

That was my only doubt. We could go with the 1st solution with `plugins: Plugin[]` then 👍